### ISSUE TITLE
UQlab,python files ,matlab files

### ISSUE THREAD

**Issue by AmosJoseph** [mabessa/F3DAS#1]
Dear Dr Bessa:

Excuse me,STEP2a_ML_Classification.py and STEP2b_ML_Regression.py are python files,but others are matlab files,then how to use them? 
Does UQlab need to be installed?

Best regards!

**Comment by mabessa** [comment]
Dear AmosJoseph,

I’m very interested in hearing your feedback on the code, as a student of mine is trying to create a new version that becomes more user-friendly and that will include tutorials. But he is just starting… So, it will take some months.

In the meantime, the code works in a very simple way:


  *   STEP1_F3DAS_supercompressible_mm.m is a Matlab code that calls Abaqus Finite Element Simulations to create a database. UQLab should be installed, but it is used just for simple tasks related to the Design of Experiments
  *   STEP2a and STEP2b are two codes to perform classification and regression, so they are used to do Machine Learning on the dataset created by STEP1. They are written in Python because I am using open-source libraries (scikit-learn, Keras, GPy and GPflow), as referenced in the 2019 Advanced Materials article.

As I mentioned, my group is currently small but it is growing. So, I expect to have improvements to make the codes more user friendly… While that doesn’t happen, feel free to contact me and I will do my best to help.

Best,
Miguel

P.S.: I am also creating a set of instructions for my students to install software that is required to do Machine Learning. I am sharing that with you below in bold.

Important Python packages to install via ANACONDA (I did this with ANACONDA for Python 2.7 version installed, not Miniconda. But it should work for that as well):

-> First create a virtual environment where you will have all the packages used for Machine Learning:

conda create -n MLenv3 python=3.6

-> Upgrade pip:

source activate MLenv3

pip install --upgrade pip

source deactivate

-> NOTE: If you are installing things like this, then you will not need to use "pip3". You use "pip" and it already knows that it is installing things on python3. I think that only if you have python2 and python3 on the same virtual environment, then you need to distinguish pip from pip3.

======================================================================================
-> Then install Tensorflow using ANACONDA (go to the website!):

*************** IGNORE OPTION 1 IF YOU WANT TO INSTALL WITH GPU **********************
- OPTION 1: Install just using CPU (no GPU):

source activate MLenv3

pip install --ignore-installed --upgrade [LINK HERE]

-> Where [LINK HERE] is the website address provided by tensorflow

source deactivate

********* If you can install CUDA, then ignore OPTION 1 and go for OPTION 2 **********
- OPTION 2 (If you are minimally experienced):

-> Install INCLUDING GPU computing, WHICH MEANS YOU NEED THE DRIVERS. SEE THE WEBSITE. For Ubuntu 16.04, this webpage explains well how to install CUDA 9.0 and cuDNN7: https://yangcha.github.io/CUDA90/

-> Now, I will include what I did here AFTER installing CUDA and other dependencies (but this can change... GO TO THE WEBSITE):

source activate MLenv3

pip install --ignore-installed --upgrade https://storage.googleapis.com/tensorflow/linux/gpu/tensorflow_gpu-1.8.0-cp36-cp36m-linux_x86_64.whl

source deactivate
**************************************************************************************
======================================================================================

-> Now take care of scikit-learn and PYGMO, including a few other things:

conda config --add channels conda-forge

conda install -n MLenv3 numpy scipy matplotlib scikit-learn pygmo

-> I also like to install a small package that deals with text labels (useful in the Pareto frontier), called AdjustText:

source activate MLenv3

pip install adjustText

-> The following

pip install h5py

-> Now install Keras (make sure you are in the MLenv virual environment):

pip install keras

-> Install genetic programming symbolic regression

pip install wheel

pip install xgp

-> I'm also starting to toy around with GPy for more advanced Gaussian Process Regression tools. NOTE: this one needs PYTHON 3 (so make sure that your conda environment is using Python3).

pip install gpy

-> And with GPflow which should be installed manually [NOTE: IT NEEDS PYTHON 3 and it does not work with python 2]:

-> To install GPflow, I cloned the GitHub repository and then (within the MLenv3 environment):

python setup.py develop

source deactivate

-> This asks for the following:

conda install -n MLenv3 mkl-service

-> VERY IMPORTANT NOTE! Strangely, if you launch SPYDER even within your virtual environment it will NOT use the packages installed in that virtual environment! This means that you need to install SPYDER inside that virtual environment, even though spyder comes by defaul with anaconda. SO, you need to do this as conda:

conda install -n MLenv3 spyder

-> Then you may want to add an alias to open spyder on the virtual environment (add following line to .bashrc or .bash_profile):

alias MLenv3_spyder='source activate MLenv3; spyder'

-> You may also want to install Sympy (to solve equations):

conda install -n MLenv3 sympy

**Comment by AmosJoseph** [comment]
OK,many thanks and I'm really excited  to receive your reply!!

**Comment by AmosJoseph** [comment]
Dear Dr Bessa:

Several questions：
1.In the paper the first and total sensitivity of A_D1 are nearly zero，then A is ignored but not the variable A_D1,why?

2. *   STEP1_F3DAS_supercompressible_mm.m.    
line189  UserSettings.postproc_compute_min = 1; % if UserSettings.postproc_compute_min = 1 it will calculate the maximum values of the requested variables       (   as is written in the comment line, is "maximum" a typing error ? )

line190  UserSettings.postproc_compute_av = 0; % if UserSettings.postproc_compute_av = 1 it will calculate the maximum values of the requested variables          (also "maximum" as is written in the comment line? )

line51  UserSettings.Input_size = 1;    (refer to line 62, there are 2 sampling points,why the value is 1?   )

3.generate_mesh_supercompressible_mm.m
line 34, what is the meaning of Twist_angle, transition_length_ratio(ratio of transition and length,then what is  transition? ) ?        what do nStories stand for?        I just donot understand the meaning of geometry representation.Would you like to help me make it clear?

Sorry to bother you！

Best regards!

**Comment by AmosJoseph** [comment]
Excuse me, what is the requirement of computer hardware running the code?
